oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java

```diff
@@ -609,26 +609,70 @@ private APIGatewayProxyResponseEvent handleAuthJourney(
         }
-        var session = existingSession.orElseGet(sessionService::generateSession);
-        attachSessionIdToLogs(session);
 
+        Session session;
+        OrchSessionItem orchSession;
+        var maxAgeParam = getMaxAge(authenticationRequest);
+        boolean isMaxAgeSupported =
+                configurationService.supportMaxAgeEnabled() && client.getMaxAgeEnabled();
+        final long timeNow = NowHelper.now().toInstant().getEpochSecond();
         Optional<String> existingSessionId = existingSession.map(Session::getSessionId);
-        if (existingSession.isEmpty()) {
-            updateAttachedSessionIdToLogs(session.getSessionId());
-            LOG.info("Created session");
+
+        if (existingSession.isEmpty() || existingOrchSessionOptional.isEmpty()) {
+            session = sessionService.generateSession();
+            var newSessionId = session.getSessionId();
+            orchSession = createNewOrchSession(newSessionId);
+            LOG.info("Created session with id: {}", newSessionId);
         } else {
-            existingSessionId = Optional.of(session.getSessionId());
-            sessionService.updateWithNewSessionId(session);
-            updateAttachedSessionIdToLogs(session.getSessionId());
-            LOG.info("Updated session id from {} - new", existingSessionId);
+            if (maxAgeParam.isPresent()
+                    && isMaxAgeSupported
+                    && maxAgeExpired(
+                            existingOrchSessionOptional.get().getAuthTime(),
+                            maxAgeParam,
+                            timeNow)) {
+                var newSessionIdForPreviousSession = IdGenerator.generate();
+                var newSessionId = IdGenerator.generate();
+                session =
+                        updateSharedSessionDueToMaxAge(
+                                existingSession.get(),
+                                newSessionIdForPreviousSession,
+                                newSessionId);
+                orchSession =
+                        updateOrchSessionDueToMaxAgeExpiry(
+                                newSessionId,
+                                existingOrchSessionOptional.get(),
+                                timeNow,
+                                newSessionIdForPreviousSession);
+
+                LOG.info(
+                        "Updated previous session ID due to max age expiry. Session ID updated from {} to {}",
+                        existingOrchSessionOptional.get().getSessionId(),
+                        newSessionIdForPreviousSession);
+
+                LOG.info(
+                        "Created new session with session ID {} due to max age expiry",
+                        orchSession.getSessionId());
+
+            } else {
+                var sessionToBeUpdated = existingSession.get();
+                sessionService.updateWithNewSessionId(sessionToBeUpdated);
+                session = sessionToBeUpdated;
```

Comment:
I'm OK with this but for readability feels a little odd to be using `sessionToBeUpdated` after it's been updated. Could replace `sessionToBeUpdated` with `session` on lines after 659

Reply:
Updated this